packages/scoped-custom-element-registry/src/scoped-custom-element-registry.js

```diff
@@ -403,4 +441,111 @@ if (!ShadowRoot.prototype.createElement) {
     configurable: true,
     writable: true,
   });
+
+  if (
+    window.ElementInternals &&
+    window.ElementInternals.prototype.setFormValue
+  ) {
+    const internalsToHostMap = new WeakMap();
+    const attachInternals = HTMLElement.prototype.attachInternals;
+    const methods = [
+      'setFormValue',
+      'setValidity',
+      'checkValidity',
+      'reportValidity',
+    ];
+
+    HTMLElement.prototype.attachInternals = function (...args) {
+      const internals = attachInternals.call(this, ...args);
+      internalsToHostMap.set(internals, this);
+      return internals;
+    };
+
+    methods.forEach((method) => {
+      const originalMethod = window.ElementInternals.prototype[method];
+
+      window.ElementInternals.prototype[method] = function (...args) {
+        const host = internalsToHostMap.get(this);
+        const definition = definitionForElement.get(host);
+        if (definition?.formAssociated === true) {
+          originalMethod.call(this, ...args);
+        } else {
+          throw new DOMException(
+            `Failed to execute ${originalMethod} on 'ElementInternals': The target element is not a form-associated custom element.`
+          );
+        }
+      };
+    });
+
+    // Emulate the native RadioNodeList object
+    class RadioNodeList extends Array {
+      constructor(elements) {
+        super(...elements);
+        this._elements = elements;
+      }
+
+      get value() {
+        return this._elements.find((element) => element.checked.value);
```

Comment:
I think you want to dereference value off of the found checked one, right? And looks like it defaults to `''` when there's nothing selected.
```suggestion
        return this._elements.find((element) => element.checked)?.value ?? '';
```

Reply:
And would you mind adding a test for the radio group case? Looks like this would have thrown.

Reply:
Added.